src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -0,0 +1,485 @@
+                    else
+                    {
+                        analyzerBuilder[key] = value;
+                    }
+                }
+            }
+        }
+
+        private static DiagnosticDescriptor InvalidAnalyzerConfigSeverityDescriptor
+            = new DiagnosticDescriptor(
+                "InvalidSeverityInAnalyzerConfig",
+                CodeAnalysisResources.WRN_InvalidSeverityInAnalyzerConfig_Title,
+                CodeAnalysisResources.WRN_InvalidSeverityInAnalyzerConfig,
+                "AnalyzerConfig",
+                DiagnosticSeverity.Warning,
+                isEnabledByDefault: true);
+
+        public Section GlobalSection { get; }
+
+        /// <summary>
+        /// The directory the editorconfig was contained in, with all directory separators
+        /// replaced with '/'.
+        /// </summary>
+        public string NormalizedDirectory { get; }
+
+        /// <summary>
+        /// The path passed to <see cref="AnalyzerConfig.Parse(string, string)"/> during construction.
+        /// </summary>
+        public string PathToFile { get; }
+
+        /// <summary>
+        /// Comparer for sorting <see cref="AnalyzerConfig"/> files by <see cref="NormalizedDirectory"/> path length.
+        /// </summary>
+        internal static Comparer<AnalyzerConfig> DirectoryLengthComparer { get; } = Comparer<AnalyzerConfig>.Create(
+            (e1, e2) => e1.NormalizedDirectory.Length.CompareTo(e2.NormalizedDirectory.Length));
+
+        public ImmutableArray<Section> NamedSections { get; }
+
+        private AnalyzerConfig(
+            Section globalSection,
+            ImmutableArray<Section> namedSections,
+            string pathToFile)
+        {
+            GlobalSection = globalSection;
+            NamedSections = namedSections;
+            PathToFile = pathToFile;
+
+            // Find the containing directory and normalize the path separators
+            string directory = Path.GetDirectoryName(pathToFile) ?? pathToFile;
+            NormalizedDirectory = PathUtilities.DirectorySeparatorChar == '/'
+                ? directory
+                : directory.Replace(PathUtilities.DirectorySeparatorChar, '/');
+        }
+
+        /// <summary>
+        /// Gets whether this editorconfig is a the topmost editorconfig.
+        /// </summary>
+        public bool IsRoot => GlobalSection.Properties.TryGetValue("root", out string val) && val == "true";
+
+        /// <summary>
+        /// Parses an editor config file text located at the given path. No parsing
+        /// errors are reported. If any line contains a parse error, it is dropped.
+        /// </summary>
+        public static AnalyzerConfig Parse(string text, string pathToFile)
+        {
+            if (!Path.IsPathRooted(pathToFile) || string.IsNullOrEmpty(Path.GetFileName(pathToFile)))
+            {
+                throw new ArgumentException("Must be an absolute path to an editorconfig file", nameof(pathToFile));
```

Comment:
This is now a public API, do we need to make this message a resource?

Reply:
Most of the ArgumentExceptions with messages in Core and CSharp use localized strings. Consider using a resource.

---
In reply to: [228340378](https://github.com/dotnet/roslyn/pull/30752#discussion_r228340378) [](ancestors = 228340378)

Reply:
@jaredpar How do you want to proceed here?

Reply:
Lets not block this PR but start an email so that we can understand what the right path is here.

---
In reply to: [228697324](https://github.com/dotnet/roslyn/pull/30752#discussion_r228697324) [](ancestors = 228697324)